impls/monero.rs/scripts/build_monero_c.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+set -x -e
+
+# See https://github.com/MrCyjaneK/monero_c for the most up-to-date build docs,
+# this is an example and a starting point for building monero_c for use in Rust
+# but it should be automated either using CMake or Cargo (preferred).
+
+# Detect architecture.
+ARCH=$(uname -m)
+OS=$(uname -s)
+
+case $ARCH-$OS in
+    x86_64-Linux)
+        TARGET_ARCH="x86_64-linux-gnu"
+        ;;
+    i686-Linux)
+        TARGET_ARCH="i686-linux-gnu"
+        ;;
+    aarch64-Linux)
+        TARGET_ARCH="aarch64-linux-gnu"
+        ;;
+    x86_64-Android)
+        TARGET_ARCH="x86_64-linux-android"
+        ;;
+    i686-Android)
+        TARGET_ARCH="i686-linux-android"
+        ;;
+    aarch64-Android)
+        TARGET_ARCH="aarch64-linux-android"
+        ;;
+    armv7l-Android)
+        TARGET_ARCH="arm-linux-androideabi"
+        ;;
+    i686-Windows)
+        TARGET_ARCH="i686-w64-mingw32"
+        ;;
+    x86_64-Windows)
+        TARGET_ARCH="x86_64-w64-mingw32"
+        ;;
+    x86_64-Darwin)
+        TARGET_ARCH="host-apple-darwin"
+        ;;
+    arm64-Darwin)
+        TARGET_ARCH="host-apple-ios"
+        ;;
+    *)
+        echo "Unsupported architecture: $ARCH on OS: $OS"
+        exit 1
+        ;;
+esac
+
+#../prepare_moneroc.sh
+# See https://github.com/cypherstack/flutter_libmonero/blob/main/scripts/prepare_moneroc.sh
+# flutter_libmonero/scripts/prepare_moneroc.sh:
+
+if [[ ! -d "monero_c" ]];
+then
+    #rm -rf monero_c
+    git clone https://github.com/sneurlax/monero_c --branch rust
+fi
+cd monero_c
+#git checkout "6eb571ea498ed7b854934785f00fabfd0dadf75b"
+git reset --hard
+#git config submodule.libs/wownero.url https://git.cypherstack.com/Cypher_Stack/wownero
+#git config submodule.libs/wownero-seed.url https://git.cypherstack.com/Cypher_Stack/wownero-seed
+git submodule update --init --force --recursive
+./apply_patches.sh monero
+#./apply_patches.sh wownero
+
+if [[ ! -f "monero/.patch-applied" ]];
+then
+    ./apply_patches.sh monero
+fi
+
+#if [[ ! -f "wownero/.patch-applied" ]];
+#then
+#    ./apply_patches.sh wownero
+#fi
+
+# flutter_libmonero/scripts/linux/build_all.sh cont. ...
+
+pushd ../monero_c
+    ./build_single.sh monero "$TARGET_ARCH" -j$(nproc)
+    #./build_single.sh wownero "$TARGET_ARCH" -j$(nproc)
+popd
```

Comment:
We should assume that monero_c was already built and is in the release/ directory, unless there is a specific reason to rebuild it for rust? So if this is not strictly required for rust it shouldn't be built at all

```suggestion
```

Reply:
It is not strictly required for Rust, this was just a hack to provide a quick way to build the exact .so I'm using.  I would prefer that no building takes place at this level whatsoever

I can get it from release/ no problem, one min.

I just wonder how this will work for projects wanting to use these Rust bindings.  Will they have to include the entire monero_c repo in their project?  (Right now yes as I use bindgen to bind ../../monero_libwallet2_api_c/src/main/cpp/wallet2_api_c.h, tho there's no reason this single file couldn't be included in the crate or handled otherwise in order for it to be standalone)

Right now, consumer crates just need the lib adjacent to the binary or in ./lib relative to the crate manifest directory.  I can add ../../release as a candidate for lib paths tho, it'll be the first lib location candidate...

Reply:
a81f173168ade4ae3f3c6d2ee38d69bec9a7a2e8 adds ../../release as the first lib location candidate but doesn't remove the script.  I can just point to monero_c/README.md for building but I'd prefer some script to be available, perhaps I just need to look harder >.>

"build it according to docs or download the release from https://github.com/MrCyjaneK/monero_c/releases" might be enough.

Reply:
Well yeah, my "dream" was to have monero_c as a system library - but that didn't happen, so for now it is sadly "just figure out what works for you", that being said there are some standard locations to put the file in.

As for the docs I'll push updated docs today - up to this PR there were a total of 4 people building this lib, and I never updated the docs :p. Let me fix that real quick.

Reply:
As for the monero_c inclusion.. that's the only way in which we could guarantee that the files are indeed correct, I am aware that this adds some overhead (and I'm sure that nobody in the flutter family is especially happy to pull C#, TS or (soon) rust code), but the way it was before, where monero_c was standalone project and monero.dart was a separate standalone project caused too many issues and created a weird update situation - where some things were updated in monero.dart, but also required monero_c to get updated (like the ledger support that @konstantinullrich is working on).

Also this caused weird dependencies to happen, for example late monero.dart v0.18.3.3 is still fully compatible with early v0.18.3.4, but breaks with later v0.18.3.4 releases... and it breaks in worst possible way - at runtime (due to different function signatures).

With a single monorepo it is easier - you have just one single git hash or tag that you need to import, (sure you can still do a version mismatch [but it's harder](https://github.com/MrCyjaneK/monero_c/blob/master/generate_checksum.sh)). If the current approach is problematic for cargo then we can figure something else out, but I believe that there are too many positive aspects and the tradeoff is essentially using a bit extra storage.